firmware/defmt-semihosting/src/lib.rs

```diff
@@ -37,6 +37,13 @@ unsafe impl defmt::Logger for Logger {
         unsafe { ENCODER.start_frame(do_write) }
     }
 
+    unsafe fn flush() {
+        // Do nothing.
+        //
+        // In QEMU write should auto-flush, because the target is emulated.
+        // With hardware write blocks for several hundred of milliseconds which should be enough to flush the data.
```

Comment:
I think the real reason this can be a no-op is that semihosting fundamentally is blocking, and does not have I/O buffers the target can control (ie. no "flush" syscall, just "write"). After write returns, the host has the data, so there's nothing left to flush.

If the semihosting transport used an internal buffer things would be different, of course.